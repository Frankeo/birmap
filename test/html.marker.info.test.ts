import {
  type FeaturePlace,
  getHtmlMarkerInfo,
} from "../src/components/map/html-marker-info";
import { expect, test, describe } from "vitest";

describe("getHtmlMarkerInfo Suite", () => {
  const title = "SOME TITLE";
  const slug = "abirradero";

  test("should return a LINK with a TITLE and NO icons when there are NOT features", async () => {
    const featuresPlace: FeaturePlace[] = [];
    const htmlMarkerInfo = await getHtmlMarkerInfo(featuresPlace, slug, title);
    expect(htmlMarkerInfo).toMatchSnapshot();
    expect(htmlMarkerInfo).toContain(slug);
    expect(htmlMarkerInfo).toContain(title);
    expect(htmlMarkerInfo).not.toContain("icon");
  });

  test("should return a LINK with a TITLE and some icons when there are some features", async () => {
    const value1 = 1;
    const value2 = 11;
    const getFeature = (value: number): FeaturePlace => ({
      feature: {
        collection: "features",
        slug: "drafts",
      },
      value,
    });
    const featuresPlace: FeaturePlace[] = [
      getFeature(value1),
      getFeature(value2),
    ];
    const htmlMarkerInfo = await getHtmlMarkerInfo(featuresPlace, slug, title);
    expect(htmlMarkerInfo).toMatchSnapshot();
    expect(htmlMarkerInfo).toContain(slug);
    expect(htmlMarkerInfo).toContain(title);
    expect(htmlMarkerInfo).toContain("icon");
    expect(htmlMarkerInfo).toContain(value1);
    expect(htmlMarkerInfo).toContain(value2);
  });
});
