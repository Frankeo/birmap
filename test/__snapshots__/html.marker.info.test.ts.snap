// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getHtmlMarkerInfo Suite > should return a LINK with a TITLE and NO icons when there are NOT features 1`] = `
"        
  <a href=\\"/places/abirradero\\" class=\\"has-text-primary\\">
    <p>SOME TITLE</p>
    
  </a>
  "
`;

exports[`getHtmlMarkerInfo Suite > should return a LINK with a TITLE and some icons when there are some features 1`] = `
"        
  <a href=\\"/places/abirradero\\" class=\\"has-text-primary\\">
    <p>SOME TITLE</p>
    <span
  style=\\"margin: 0;\\"
  class=\\"icon-text has-text-primary has-text-weight-bold\\"
>
  <span>1</span>
  <span class=\\"icon\\">
    <i class=\\"fa-solid fa-beer-mug-empty \\"></i>
  </span>
</span><span
  style=\\"margin: 0;\\"
  class=\\"icon-text has-text-primary has-text-weight-bold\\"
>
  <span>11</span>
  <span class=\\"icon\\">
    <i class=\\"fa-solid fa-beer-mug-empty \\"></i>
  </span>
</span>
  </a>
  "
`;
